### DISCUSSION TITLE
Airmon-ng error with mount sysfs

### DISCUSSION THREAD

**Discussion by gyhxrr31** [aircrack-ng/aircrack-ng#2486] [Q&A] [answered]
I am getting an error with airmon-ng on my redmi note 7 device with custom kernel and rom. And the error is only on my kernel, which I myself compiled from source.  This error ONLY occurs on my kernel. When I try to mount, it says that the partition is already mounted: `mount: /sys: sysfs already mounted or mount point busy. call. dmesg(1) may have more information after failed mount system`. I've already tried a lot, but I don't know what to do anymore. 
```
ERROR: airmon-ng wlan2 start
/sys is not mounted, you can probably mount it with: mount -t sysfs sysfs /sys
This program cannot continue without a working sysfs.
```

**Comment by Pavis0247** [comment]
> I am getting an error with airmon-ng on my redmi note 7 device with custom kernel and rom. And the error is only on my kernel, which I myself compiled from source.  This error ONLY occurs on my kernel. When I try to mount, it says that the partition is already mounted: `mount: /sys: sysfs already mounted or mount point busy. call. dmesg(1) may have more information after failed mount system`. I've already tried a lot, but I don't know what to do anymore. 
> ```
> ERROR: airmon-ng wlan2 start
> /sys is not mounted, you can probably mount it with: mount -t sysfs sysfs /sys
> This program cannot continue without a working sysfs.
> ```

Did you get a solution?

**Reply by Mister-X-** [reply]
No. And people would have responded if they had.

We don't have that specific hardware so we're unable to test or reproduce the issue

**Reply by gyhxrr31** [reply, marked as answer]
I simply deleted in the airmon-ng script (I think I don’t remember exactly, because I did it a long time ago) the lines related to checking the sysfs mount and it is fixed. 
![ss1](https://github.com/user-attachments/assets/f8c70f9e-2162-4794-ad4f-0d78e0790382)
![ss2](https://github.com/user-attachments/assets/9f89a044-5a41-410a-a162-84ecca66bd69)

**Comment by naivor** [comment]
i meet the same problem on openwrt 23.05, and ```mount sysfs sysfs /sys ``` not work ,it shows  resource busy

**Reply by RobertAzovski** [reply]
> i meet the same problem on openwrt 23.05, and `mount sysfs sysfs /sys ` not work ,it shows resource busy

did you find a solution ?

**Reply by gyhxrr31** [reply]
> > i meet the same problem on openwrt 23.05, and `mount sysfs sysfs /sys ` not work ,it shows resource busy
> 
> did you find a solution ?

Simply edit airmon-ng script, delete function which is checking sysfs

**Comment by fajil1269-boop** [comment]
<img width="1080" height="1920" alt="Screenshot_20250918-222756" src="https://github.com/user-attachments/assets/6c379901-5643-4020-83f8-f21bf391b5ef" />
Where will i get interface?
